### DISCUSSION TITLE
`parser` configuration inconsistency (vs plugins) in a TypeScript monorepo

### DISCUSSION THREAD

**Discussion by alexrp** [eslint/eslint#17092] [Q&A] [answered]
#12949 was raised a few years ago asking why `@typescript-eslint/parser` can't be configured as `parser` in the root `.eslintrc.json` in a monorepo where `@typescript-eslint/parser` is installed in a sub-package. It was closed with this statement:

> We are going in the opposite direction -- ESLint loads plugins as well from the directory of each config file that declares the plugin since ESLint 7.

However, empirically, this doesn't seem to be the case with ESLint 8.38.0. I am able to configure the `@typescript-eslint` plugin in my root `.eslintrc.json` just fine, while having it installed in a sub-package. It also runs as expected, flagging code that it should. But `@typescript-eslint/parser` *has* to be configured in the sub-package's `.eslintrc.json`.

Am I just missing something here? 🤔 It seems to me that #12949 was a valid issue pointing out an odd configuration inconsistency.

**Comment by nzakas** [comment]
You'd need to ask the folks over at  [typescript-eslint](https://github.com/typescript-eslint/typescript-eslint)

**Reply by alexrp** [reply]
Just to be clear, does that mean loading of the parser is handled by the plugin?

**Reply by Rec0iL99** [reply]
I think ESLint does the loading of the parser.

**Reply by alexrp** [reply]
Then isn't this repo the right place for the question? Sorry if I'm way off, I'm happy to go wherever is appropriate, I'm just a bit confused at the moment. 😅

**Reply by nzakas** [reply]
ESLint loads the parser, but how you configure the parser is up to the parser itself. We can't comment on the right way to install the typescript-eslint parser, so you're better off asking those folks.

**Reply by alexrp** [reply]
I think I might have made some poor word choices in the OP. To be clear, I'm not talking about configuration options *for* the parser, but rather, I have ~this `.eslintrc.json` in the sub-package:

```json
{
  "$schema": "https://json.schemastore.org/eslintrc.json",
  "extends": "../../../.eslintrc.json",
  "parser": "@typescript-eslint/parser"
}
```

If I move that `parser` key to the root `.eslintrc.json`, it fails to load as described above and in #12949. So we're talking about an issue with loading the parser in the first place, not with configuring it.

**Reply by nzakas** [reply, marked as answer]
Ah okay. So yes, loading parsers in shareable configs gets complicated because of the way packages are resolved in eslintrc, this is part of why we're moving to a new config system. These types of problems are just bears to try and solve, so unfortunately, we can't do anything about it right now. I'd suggest investigating switching your project to the new config system (called flat config), which will make your life a lot easier.
